### PR TITLE
Add TransformComponent and Matrix

### PR DESCRIPTION
**Pull request by nattridgell**
Adds:
- TransformComponent
- Very basic TransformComponent tests
- Matrix (for transforms)
- Matrix.multiply functions for multiplying 2D and 3D matrices
  Note: Will be needed by TransformValue
- Matrix.is2DComponent function
- many tests for matrix multiplication using
  precomputed results

### CODE REVIEW

File: src/matrix.js [hunk 1]
```diff
@@ -0,0 +1,121 @@
...
+  // Note: A Matrix is stored and referenced in column-major order.
+  // TODO: Update the names of parameters (a, b, ... , p) once they are changed
+  // in the specifications and add getters for these parameters.
+  function Matrix(a, b, c, d, e, f, g, h, i, j, k, l, m, n, o, p) {
+    if (arguments.length != 6 && arguments.length != 16) {
+      throw new TypeError('Expecting 6 or 16 arguments.');
+    }
+
+    this._matrix = [];
+    for (var index = 0; index < arguments.length; index++) {
```
Comment (wilddamon) [inline src/matrix.js:26]: It's more common just to use "i" as the variable name here.

File: src/transform-component.js [hunk 2]
```diff
@@ -0,0 +1,35 @@
...
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+//     See the License for the specific language governing permissions and
+// limitations under the License.
+
+(function(internal, testing) {
+
+  function TransformComponent() {
+  }
+
+  TransformComponent.prototype = {
```
Comment (wilddamon) [inline src/transform-component.js]: I think we were moving to get rid of inline declarations like these... instead please use

TransformComponent.prototype.asMatrix = function() { ... }
TransformComponent.prototype.isDDComponent = function() { ... }

File: test/js/transform-component.js [hunk 3]
```diff
@@ -0,0 +1,13 @@
+suite('TransformComponent', function() {
+  test('TransformComponent is a TransformComponent', function() {
```
Comment (wilddamon) [inline test/js/transform-component.js]: This seems redundant... no need to test language features?

File: src/matrix.js (hunk 1 above)
Comment (nattridgell) [inline src/matrix.js:26]: Cannot, i, j, and k are defined as parameters.

**Comment by wilddamon** [comment]
LGTM

**Pull request merged**